_posts/2024-08-14-running-a-hsm-realm.md

```diff
@@ -0,0 +1,221 @@
+---
+layout: post
+title: "Running a Juicebox HSM realm"
+description: "Details on how to setup and run a Juicebox HSM realm with Entrust HSMs."
+date: 2024-08-14 11:30:00 -0700
+author-name: Simon Fell
+author-github: superfell
+---
+
+By necessity the architecture of the [HSM
```

Comment:
I wonder if we can give a little more framing at the start, maybe changing this intro section to something like:

As we [previously mentioned](https://juicebox.xyz/blog/key-to-simplicity-squeezing-the-hassle-out-of-encryption-key-recovery#trusting-deliberately), the Juicebox protocol allows for two types of realms – **_hardware realms_** backed by secure hardware (HSMs), and **_software realms_** that can run on commodity hardware. This blog post aims to provide a deeper look into the process of running the former.

If you've been looking at the source code we published, you may have noticed the architecture of our [hardware realm](https://github.com/juicebox-systems/juicebox-hsm-realm) is more complex than our [software realm](https://github.com/juicebox-systems/juicebox-software-realm). This is by design, as it requires multiple HSMs, physical servers to host them, and cloud resources to provide its significantly higher level of security. Additionally, these realms require a lengthy initialization and deployment process that software realms avoid. 

As you make your way through this blog post, you may want to reference the useful information located in the [HSM realm readme](https://github.com/juicebox-systems/juicebox-hsm-realm/blob/main/README.md) and the [Entrust HSM module readme](https://github.com/juicebox-systems/juicebox-hsm-realm/blob/main/entrust_hsm/README.md).